pkg/extension/opampextension/opamp_agent_test.go

```diff
@@ -31,12 +31,36 @@ import (
 	semconv "go.opentelemetry.io/collector/semconv/v1.18.0"
 )
 
-func TestNewOpampAgent(t *testing.T) {
-	cfg := createDefaultConfig()
+func setupOpampAgent(t *testing.T) (*opampAgent, *Config) {
```

Comment:
I believe the goal here is to convert this to be parameterized tests in go - https://stackoverflow.com/questions/72856565/go-golang-test-parameterized

Reply:
Yes, we should follow the manner: https://github.com/SumoLogic/sumologic-otel-collector/blob/c7417c6cbeadbe7e79b3c6c2b84038c1ee30b8af/pkg/exporter/sumologicexporter/config_test.go#L12-L111

so, we have the `testcases` struct with parameters and we iterate over it, so instead of having 5 different `Test` functions, we have just 5-element list

Reply:
addressed